include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+        unset($event_remote_empty); // We're done with this
+
+        return $event_remote;
+    }
+
+    /**
+     * Set the timezone
+     *
+     * @param string $timezone : the timezone_identifier (ie. America/New_York)
+     *
+     * @return bool Success/Failure
+     */
+    public function setTimezone($timezone)
+    {
+        if (!date_default_timezone_set($timezone))
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Failed to set timezone to \'' . $timezone . '\'');
+            return false;
+        }
+        else {
+            $this->timezone = date_default_timezone_get();
+            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Timezone set to \'' . $this->timezone . '\'');
+            return true;
+        }
+    }
+
+    /**
+     * Set the last sync time for the record
+     *
+     * This *must* be called *after* the sync is done
+     * This also saves the event, so you don't need to do it twice. Just call this.
+     * Also adds the id of the event so it doesn't get synced again this session.
+     *
+     * @param object $event_local The SuiteCRM Meeting bean
+     * @param string $gEventId (optional) The ID that Google has for the event.
+     *
+     * @return bool Success/Failure
+     */
+    protected function setLastSync($event_local, $gEventId = NULL)
+    {
+
+        if (isset($gEventId))
+        {
+            $event_local->gsync_id = $gEventId;
+        }
+
+        $event_local->gsync_lastsync = time() + 3; // we add three seconds to this, so that the modified time is always older than the gsync_lastsync time
+        $return = $event_local->save(false);
+
+        // Set the meeting as accepted by the user, otherwise it doesn't show up on the calendar. We do it here because it must be saved first.
+        $event_local->set_accept_status($this->workingUser, 'accept');
+
+        if (isset($return))
+        {
+            $this->syncedList[] = $event_local->id;
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * Perform the sync for a user
+     *
+     * @param string $id The SuiteCRM user id
+     */
+    public function doSync($id)
+    {
+
+        if (!$this->setClient($id))
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to setup Google Client');
+            return false;
+        }
+
+        if ($this->workingUser->id == $id)
+        {
+            $tz = $this->workingUser->getPreference('timezone', 'global');
+            $this->setTimezone($tz);
+        }
+        else {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Failed to set the working user and timezone');
+            return false;
+        }
+
+        if (!$this->setGService($this->gClient))
```

Comment:
The signature of setGService does not take any arguments. Is this a bug? I think this line needs to be refactored.

Reply:
I missed this piece of cruft from older code. I was passing the GClient previously.